pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -673,6 +673,10 @@ def _snstop(self, ktcond, mjrprtlvl, minimize, n, nncon, nnobj, ns, itn, nmajor,
             if not self.storeHistory:
                 raise Error("snSTOP function handle must be used with storeHistory=True")
             iabort = snstop_handle(iterDict)
+            # write iterDict again if anything was inserted
```

Comment:
I don't fully understand this change. Does `iterDict` get written again or overwritten? Also, does this happen strictly if anything was inserted or are you saying write it again in case anything was inserted?

Reply:
It gets updated (in the Python dictionary sense) by the new `iterDict`, then written into the history file. There is no check so this is performed every time. What this mean is that new values/entries are written, but if the user removed an entry or something, the entry remains in the history file.